deployment/cloudformation/privatehostedzone.py

```diff
@@ -19,21 +18,32 @@ class R53PrivateHostedZone(CustomActionNode):
 
     def action(self):
         """Creates the zone"""
-        conn = route53.connect_to_region(self.REGION)
+        client = boto3.client('route53', region_name=self.REGION)
         comment = json.dumps(self.get_raw_tags())
 
-        hosted_zones = conn.get_all_hosted_zones()
+        hosted_zones = client.list_hosted_zones()
 
-        for hosted_zone in hosted_zones['ListHostedZonesResponse']['HostedZones']:
+        for hosted_zone in hosted_zones['HostedZones']:
             if ('Comment' in hosted_zone['Config'] and
                     hosted_zone['Config']['Comment'] == comment):
                 self.stack_outputs = {'PrivateHostedZoneId': hosted_zone['Id'].split('/')[-1]}
                 return
-
-        hosted_zone = conn.create_hosted_zone(self.get_input('PrivateHostedZoneName'),
-                                              comment=comment,
-                                              private_zone=True,
-                                              vpc_id=self.get_input('VpcId'),
-                                              vpc_region=self.REGION)
+            
+        vpc_config = {
+            'VPCRegion': self.REGION,
+            'VPCId': self.get_input("VpcId")
+        }
+        hosted_zone_config = {
+            'Comment': comment,
+            'PrivateZone': True
+        }
+
+        hosted_zone = client.create_hosted_zone(
+            Name=self.get_input("PrivateHostedZoneName"),
+            CallerReference=str(uuid.uuid4()),
```

Comment:
This method in `boto` had `CallerReference` (`caller_ref`) as an optional parameter and it would generate a Type 4 UUID to use by default. It is now required in `boto3` and can be any unique string, so I'm continuing to use uuid by default here.